sdk/src/wallet/account/operations/balance.rs

```diff
@@ -63,13 +63,17 @@ where
     ) -> Result<Balance> {
         let network_id = self.client().get_network_id().await?;
         let rent_structure = self.client().get_rent_structure().await?;
+        let local_time = self.client().get_time_checked().await?;
         let mut balance = Balance::default();
         let mut total_rent_amount = 0;
         let mut total_native_tokens = NativeTokensBuilder::default();
 
         #[cfg(feature = "participation")]
         let voting_output = self.get_voting_output().await?;
 
+        let account_addresses = self.addresses().await?;
+        let claimable_outputs = account_details.claimable_outputs(OutputsToClaim::All, local_time)?;
```

Comment:
this can be `self.claimable_outputs(OutputsToClaim::All)`

Reply:
Have you read the description? :trollface: Because it was like that before, it has led  to a deadlock (because internally it does `self.read()`)

Reply:
But so does self.addresses(), which isnt a problem? I suspect the real gain is not calling the lock twice per iteration time and claimable_outputs)

Reply:
True, self.addresses() can actually have the same problem and self.get_voting_output() as well
For that we would need to update also some more things that are called internally

Reply:
I think you're right on this, should call addresses on the account_details?